include/macros.h

```diff
@@ -0,0 +1,88 @@
+
+#ifndef DECAF_MACROS_H
+#define DECAF_MACROS_H
+
+#include <string_view>
+
+namespace decaf {
+  namespace detail {
+    // Optional-like type for use in message creation.
+    //
+    // I'm trying to keep this header lightweight so that including it
+    // everywhere is less of an issue.
+    struct message {
```

Comment:
is this class also supposed to be used directly? The `string_view` looks like a good place for use after free errors

Reply:
No. It's part of an implementation detail to allow the assertion macros to be called as both
```
DECAF_ASSERT(<cond>);
DECAF_ASSERT(<cond>, "some failure message");
```
In this case it shouldn't outlive the expression in which it's declared so the string_view isn't a problem.

I've gone and added some docs to clarify this.